rest/test/routes/diagnosticRoutes_spec.js

```diff
@@ -21,11 +21,72 @@
 const diagnosticRoutes = require('../../src/routes/diagnosticRoutes');
 const test = require('./utils/routeTestUtils');
 const { expect } = require('chai');
+const { version: restVersion } = require('../../package.json');
+const { version: sdkVersion } = require('../../../catapult-sdk/package.json');
 
 describe('diagnostic routes', () => {
 	const executeRoute = (routeName, db, assertResponse) =>
 		test.route.executeSingle(diagnosticRoutes.register, routeName, 'get', {}, db, undefined, assertResponse);
 
+	describe('blocks', () => {
+		const builder = test.route.document.prepareGetDocumentsRouteTests(diagnosticRoutes.register, {
+			route: '/diagnostic/blocks/:height/limit/:limit',
+			dbApiName: 'blocksFrom',
+			type: 'blockHeaderWithMetadata'
+		});
+
+		builder.addValidInputTest({ object: { height: '1234', limit: '4321' }, parsed: [1234, 4321] });
+		builder.addEmptyArrayTest({ object: { height: '1234', limit: '4321' }, parsed: [1234, 4321] });
+
+		// notice that this expands to four tests { 'height', 'limit'} x { '10A', '-4321' }
+		['height', 'limit'].forEach(property => ['10A', '-4321'].forEach(value => {
+			const object = Object.assign({ height: '1234', limit: '4321' }, { [property]: value });
+			const errorMessage = `${property} has an invalid format`;
+			builder.addInvalidKeyTest({ object, error: errorMessage }, `(${property} with value ${value})`);
+		}));
+	});
+
+	describe('server info', () => {
+		it('can retrieve info', done => {
+			// Arrange:
+			const endpointUnderTest = '/diagnostic/serverInfo';
+			const routes = {};
+			const server = {
+				get: (path, handler) => {
+					routes[path] = handler;
+				}
+			};
+
+			diagnosticRoutes.register(server, {});
+
+			let sentResponse;
+			const next = () => {};
+			const res = {
+				send: response => {
+					sentResponse = response;
+				},
+				redirect: () => {
+					next();
+				}
+			};
```

Comment:
should really refactor these eventually ^^

Reply:
yes, this is on the list of tech debt i owe you 😅